src/Git/GitGraphUpdater.cpp

```diff
@@ -0,0 +1,506 @@
+#include "Git/GitGraphUpdater.h"
+
+#include "Data/Graph.h"
+#include "Data/Node.h"
+#include "Data/Edge.h"
+
+#include "Manager/Manager.h"
+
+#include "Git/GitEvolutionGraph.h"
+#include "Git/GitFile.h"
+#include "Git/GitType.h"
+#include "Git/GitVersion.h"
+
+#include <QDebug>
+#include <QMapIterator>
+
+
+Git::GitGraphUpdater::GitGraphUpdater( int currentVersion, Git::GitEvolutionGraph* evolutionGraph, Data::Graph* activeGraph ) : currentVersion( currentVersion ), evolutionGraph( evolutionGraph ), activeGraph( activeGraph )
+{
+
+	// Inicializacia spracovanie cez viacere verzie
+	this->moreVersionChanged = false;
+}
+
+void Git::GitGraphUpdater::nextVersion()
+{
+	// zistime zmenene subory v dalsej verzii, s ktorymi budeme dalej pracovat
+	QList<Git::GitFile*> gitFiles = this->getEvolutionGraph()->getVersion( this->getCurrentVersion() + 1 )->getChangedFiles();
+	Data::Node* lAuthorNode = nullptr;
+	QString lAuthorName = nullptr;
+
+	// Ak ide o zmenu len cez jednu verziu, tak vymazem hrany od autora a uzol autora z grafu a pridam autora dalsej verzie
+	if ( !moreVersionChanged ) {
+		// Vymazanie hran od autora a vymazanie autorovho uzla
+		removeAuthorEdgesFromGraph( this->getEvolutionGraph()->getVersion( this->getCurrentVersion() )->getAuthor(), this->getEvolutionGraph()->getVersion( this->getCurrentVersion() )->getChangedFiles() );
+		this->getActiveGraph()->removeNode( this->getActiveGraph()->findNodeByName( this->getEvolutionGraph()->getVersion( this->getCurrentVersion() )->getAuthor() ) );
+
+		// Ziskame meno autora
+		lAuthorName = this->getEvolutionGraph()->getVersion( this->currentVersion + 1 )->getAuthor();
+
+		lAuthorNode = this->getActiveGraph()->findNodeByName( lAuthorName );
+		// Ak autor neexistuje v grafe, tak ho vytvorim
+		if ( !lAuthorNode ) {
+			lAuthorNode = this->getActiveGraph()->addNode( lAuthorName, this->getActiveGraph()->getTypesByName( "author" ).at( 0 ) );
+			lAuthorNode->setLabelText( lAuthorNode->Data::AbsNode::getName() );
+			lAuthorNode->showLabel( true );
+		}
+	}
+
+	bool ok = true;
+
+	// Pre kazdy zmeneny subor vo verzii, zistim jeho ciastocne cesty a podla typu suboru, vykonam akciu nad tymito cestami
+	foreach ( Git::GitFile* gitFile, gitFiles ) {
+		QStringList lList = gitFile->getFilepath().split( "/" );
+		QString lPom = "";
+		bool debug = false;
+		if ( gitFile->getFilepath() == "" ) {
+			debug = true;
+		}
+
+		for ( int i = 0; i < lList.size(); i++ ) {
+			lPom += lList.at( i );
+			lList.replace( i, lPom );
+			lPom += "/";
+		}
+		switch ( gitFile->getType() ) {
+			// Ak bol pridany v dalsej verzii, tak ho pridame do grafu
+			case Git::GitType::ADDED :
```

Comment:
Git::GitType::NONE sa nemôže pre triedu GitFile nastaviť. Zbytočný nie je využíva sa v triede Git FileDiffBlockLine. Je to riešené takto z dôvodu, že súčasný projekt nepodporuje enum class. Pridám tam default case.